test/unit/view.js

```diff
@@ -601,6 +601,232 @@ describe('View class', () => {
         renderStub.restore();
       });
     });
+
+    describe('getters', () => {
+      describe('outerHeight', () => {
+        let getPropertyValueSpy;
+
+        beforeEach(() => {
+          view.wrapper = document.createElement('div');
+          getPropertyValueSpy = sinon.spy(CSSStyleDeclaration.prototype, 'getPropertyValue');
+        });
+
+        afterEach(() => {
+          getPropertyValueSpy.restore();
+        });
+
+        it('returns wrapper height if there is no styling on wrapper', () => {
+          view.wrapper = {clientHeight: 10};
+          const getComputedStyleStub = sinon.stub(window, 'getComputedStyle').returns(null);
+          assert.equal(view.outerHeight, '10px');
+          getComputedStyleStub.restore();
+        });
+
+        it('returns the height of the wrapper set in style', () => {
+          const getComputedStyleStub = sinon.stub(window, 'getComputedStyle').returns(view.wrapper.style);
+          view.wrapper.style.height = '50px';
+          assert.equal(view.outerHeight, '50px');
+          getComputedStyleStub.restore();
+        });
+
+        it('returns wrapper\'s client height if height is not set in style', () => {
+          view.wrapper = {
+            clientHeight: 20,
+            style: {
+              height: '',
+              getPropertyValue: sinon.spy(),
+            },
+          };
+          const getComputedStyleStub = sinon.stub(window, 'getComputedStyle').returns(view.wrapper.style);
+          assert.equal(view.outerHeight, '20px');
+          getComputedStyleStub.restore();
+        });
+
+        it('calls getPropertyValue() twice if height is not set in style', () => {
+          view.wrapper.style.height = '';
+          assert.equal(view.outerHeight, `${view.wrapper.clientHeight}px`);
+          assert.calledTwice(getPropertyValueSpy);
+          assert.calledWith(getPropertyValueSpy, 'height');
+        });
+
+        it('calls getPropertyValue() twice if height is set to 0px in style', () => {
+          view.wrapper.style.height = '0px';
+          assert.equal(view.outerHeight, '0px');
+          assert.calledTwice(getPropertyValueSpy);
+          assert.calledWith(getPropertyValueSpy, 'height');
+        });
+
+        it('calls getPropertyValue() twice if height is set to auto in style', () => {
+          view.wrapper.style.height = 'auto';
+          assert.equal(view.outerHeight, '0px');
+          assert.calledTwice(getPropertyValueSpy);
+          assert.calledWith(getPropertyValueSpy, 'height');
+        });
+
+        it('calls getPropertyValue() once if height is set to a value in style', () => {
+          const getComputedStyleStub = sinon.stub(window, 'getComputedStyle').returns(view.wrapper.style);
```

Comment:
Why we need this stub here, but not in the few tests above?

Reply:
The stub is used to grab the view.wrapper.style.height in the line below because the original function does not read it properly

Reply:
If we were to stub out `getComputedStyle` in the tests above, would be able to use `assert.equal(...)` instead of `assert.isOk`?

Reply:
Yep!